controllers/registry.go

```diff
@@ -344,19 +390,34 @@ func getRegistryImage(dpa *oadpv1alpha1.DataProtectionApplication) string {
 	return fmt.Sprintf("%v/%v/%v:%v", os.Getenv("REGISTRY"), os.Getenv("PROJECT"), os.Getenv("VELERO_REGISTRY_REPO"), os.Getenv("VELERO_REGISTRY_TAG"))
 }
 
-func (r *DPAReconciler) buildRegistryContainer(bsl *velerov1.BackupStorageLocation, dpa *oadpv1alpha1.DataProtectionApplication) ([]corev1.Container, error) {
+func (r *DPAReconciler) buildRegistryContainer(bsl *velerov1.BackupStorageLocation, dpa *oadpv1alpha1.DataProtectionApplication, containerPort *int32) ([]corev1.Container, error) {
+	if containerPort == nil {
+		containerPort = pointer.Int32(5000)
+	}
+	if *containerPort < 5000 || *containerPort > 65534 {
+		return nil, fmt.Errorf("invalid container port %v. has to be 5000 < x < 65535", containerPort)
+	}
+	debugPort := *containerPort + 1
 	envVars, err := r.getRegistryEnvVars(bsl)
 	if err != nil {
 		r.Log.Info(fmt.Sprintf("Error building registry container for backupstoragelocation %s/%s, could not fetch registry env vars", bsl.Namespace, bsl.Name))
 		return nil, err
 	}
+	// append port config to env vars
+	envVars = append(envVars, corev1.EnvVar{
+		Name: "REGISTRY_HTTP_ADDR",
+		Value: fmt.Sprintf(":%v", *containerPort),
```

Comment:
Alternative setup that will only allow localhost access. This would work with velero container but will break Liveness/Readiness probe.
```suggestion
		Value: fmt.Sprintf("localhost:%v", *containerPort),
```

Reply:
As setup, without explicit localhost binding, pod may still accessible by knowing pod ip address from other internal workloads.

Reply:
Looks like liveness probe issue can be resolved by switching to by curling inside the pod.
```
curl --fail 10.130.2.90:5000/v2/_catalog?n=5
```

Reply:
verified fix in c4c804d
![image](https://user-images.githubusercontent.com/11228024/168725580-e40fbcd0-2aa0-4176-b6cd-17531c2f6f9c.png)
![image](https://user-images.githubusercontent.com/11228024/168725655-c265412e-9189-47c5-979d-04245ac460ab.png)